core/deployment/src/main/java/io/quarkus/deployment/builditem/LogSocketFormatBuildItem.java

```diff
@@ -0,0 +1,36 @@
+package io.quarkus.deployment.builditem;
+
+import java.util.Optional;
+import java.util.logging.Formatter;
+
+import org.wildfly.common.Assert;
+
+import io.quarkus.builder.item.MultiBuildItem;
+import io.quarkus.runtime.RuntimeValue;
+
+/**
+ * The socket format build item. Producing this item will cause the logging subsystem to disregard its
+ * socket logging formatting configuration and use the formatter provided instead. If multiple formatters
+ * are enabled at runtime, a warning message is printed and only one is used.
+ */
+public final class LogSocketFormatBuildItem extends MultiBuildItem {
```

Comment:
Won't this break if we have more than one socket handler configured for some reason?

Reply:
you mean that we should be able to work with "named" handlers? To allow writing logs to 2 different socket appenders? (Logstash and Splunk for instance)

I must admit I don't know. This is just a copy of `LogFileFormatBuildItem` for instance